cmd/rewards-exporter/main.go

```diff
@@ -82,29 +82,35 @@ func main() {
 				logrus.Fatal(err)
 			}
 			if int64(head.FinalizedEpoch) <= *epoch {
-
+				logrus.Infof("pausing %v <= %v", int64(head.FinalizedEpoch), *epoch)
 				services.ReportStatus("rewardsExporter", "Running", nil)
 				time.Sleep(time.Second * 12)
 				continue
 			}
 
-			*epoch = int64(head.FinalizedEpoch)
+			if *epoch == -1 {
+				*epoch = int64(head.FinalizedEpoch) - 1
+			}
 
-			start := time.Now()
-			logrus.Infof("retrieving rewards details for epoch %v", *epoch)
+			for i := *epoch + 1; i <= int64(head.FinalizedEpoch); i++ {
```

Comment:
I don't understand this for loop. If I read this correctly, this for loop will ALWAYS only be executed once (for `i = head.FinalizedEpoch`):

We only get here if the exporter is started with `*epoch` being set to -1. And let's say that the current `head.FinalizedEpoch` is 100.

So in the first iteration of the endless loop (line 79), `*epoch` will be be set to 99 (see line 92).
The for loop therefore goes from 100 (`*epoch + 1`) to `<= 100`: It will be executed exactly once for epoch 100 which is `head.FinalizedEpoch`.

Once that is done, `*epoch` is set to 100 (line 113) and the exporter waits until `head.FinalizedEpoch` is set to 101 (line 80 and if in line 84).

The new for loop therefore now starts with `i` being set to 101 (remember, `*epoch` is still 100) and loops exactly once again. This behavior then continues from here.

If all that is correct, I suggest to remove the for loop.

Reply:
As discussed, `head.FinalizedEpoch` might be incremented by more than 1 and the for loop ensures that nothing is missed. I will resolve this conversation.